pkg/controller/util/endpoint/controller_utils.go

```diff
@@ -281,11 +281,11 @@ func (sl portsInOrder) Less(i, j int) bool {
 // but excludes equality checks that would have already been covered with
 // endpoint hashing (see hashEndpoint func for more info).
 func EndpointsEqualBeyondHash(ep1, ep2 *discovery.Endpoint) bool {
-	if stringPtrChanged(ep1.NodeName, ep1.NodeName) {
+	if stringPtrChanged(ep1.NodeName, ep2.NodeName) {
```

Comment:
cc: @robscott

Reply:
great catch :smiling_face_with_tear:

Reply:
Created PR #108123 for fixing it in case separating PRs for different purposes is prefered. The commit in #108123 is also included in this PR.